syntax = "proto3";

package PhoneBook;

message Person {
    string name = 1;
    int32 id = 2; // Unique ID number for this person.
    string email = 3;

    enum PhoneType {
        MOBILE = 0;
        HOME = 1;
        WORK = 2;
    }

    message PhoneNumber {
        required string number = 1; // phone number is always required
        PhoneType type = 2;
    }
    // all registered phones
    repeated PhoneNumber phones = 4;
}
